Modules/Loadable/Volumes/Logic/vtkSlicerVolumesLogic.cxx

```diff
@@ -1216,15 +1204,38 @@ CloneVolume (vtkMRMLScene *scene, vtkMRMLVolumeNode *volumeNode, const char *nam
     vtkErrorWithObjectMacro(volumeNode, "Could not clone volume!");
     return NULL;
     }
-
   clonedVolumeNode->CopyWithScene(volumeNode);
+
+  // remove storage nodes
   clonedVolumeNode->SetAndObserveStorageNodeID(NULL);
-  std::string uname = scene->GetUniqueNameByString(name);
-  clonedVolumeNode->SetName(uname.c_str());
-  if ( clonedDisplayNode )
+
+  // remove display nodes (but not the first one)
+  for ( unsigned int i = 1; i < clonedVolumeNode->GetNumberOfDisplayNodes(); ++i )
```

Comment:
Nitpicks: This form may be simpler:

```
while (clonedVolumeNode->GetNumberOfDisplayNodes() > 1)
  {
  clonedVolumeNode->RemoveNthDisplayNodeID(1); // always remove at index 1 since they will shift
  }
```